php/libraries/rest_api/Entrypoint.class.inc

```diff
@@ -0,0 +1,267 @@
+ */
+
+namespace LORIS\RestAPI;
+
+/**
+ * The Entrypoint is the controler of the flow of operations.
+ *
+ * PHP Version 7
+ *
+ * @category API
+ * @package  Loris
+ * @author   Xavier Lecours Boucher <xavier.lecoursboucher@mcgill.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris-Trunk/
+ */
+class Entrypoint
+{
+    /**
+     * The HTTP method this request was made in, either GET, POST, PUT or DELETE
+     */
+    private $_method = '';
+
+    /**
+     * Any additional URI components after the endpoint and verb have been removed, in our
+     * case, an integer ID for the resource. eg: /<endpoint>/<verb>/<arg0>/<arg1>
+     * or /<endpoint>/<arg0>
+     */
+    private $_args = Array();
+
+    /**
+     * Stores the input of the PUT request
+     */
+    private $_inputs = array();
+
+    public function __construct()
+    {
+        $this->_header("Access-Control-Allow-Origin: *");
+        $this->_header("Content-Type: application/json");
+
+        $this->_args = explode('/', rtrim($_REQUEST['req'], '/'));
+
+
+        if (!array_key_exists(0, $this->_args)) {
+            $this->_error("Endpoint not specified", 404);
+        }
+
+        $this->_request  = $this->_cleanInputs($this->_args);
+
+        $this->_method = $_SERVER['REQUEST_METHOD'];
+        // X-HTTP-Method header can be added to a POST request that signals that the
+        // server MUST process the request not as a POST
+        if ($this->_method == 'POST' && array_key_exists('HTTP_X_HTTP_METHOD', $_SERVER)) {
+            if ($_SERVER['HTTP_X_HTTP_METHOD'] == 'DELETE') {
+                $this->_method = 'DELETE';
+            } else if ($_SERVER['HTTP_X_HTTP_METHOD'] == 'PUT') {
+                $this->_method = 'PUT';
+            } else {
+                $this->_error("Method not recognized", 501);
+            }
+        }
+
+        switch($this->_method) {
+            case 'POST':
+            case 'PUT':
+            case 'DELETE':
+                $input = json_decode(file_get_contents("php://input"),true);
+                if (!is_array($input)) {
+                    $this->_error("Invalid input", 400);
+                }
+                $this->_inputs = $this->_cleanInputs($input);
+            case 'GET':
+            case 'OPTIONS':
+                break;
+            default:
+                $this->_error("Invalid Method $this->_method", 405);
+                break;
+        }
+
+        unset($_GET);
+        unset($_POST);
+        unset($_REQUEST);
+
+        $client = new \NDB_Client();
+
+        $client->makeCommandLine();
+        $client->initialize();
+    }
+    
+    private function _cleanInputs($data) {
+        $clean_input = Array();
+        if (is_array($data)) {
+            foreach ($data as $k => $v) {
+                $clean_input[$k] = $this->_cleanInputs($v);
+            }
+        } else {
+            $clean_input = trim($data);
```

Comment:
This may be a good place to urlencode the values aswell. @driusan what are your thoughts?